### PR TITLE
Add a parameter for randomized password

### LINKED ISSUE #187: Remove hardcoded passwords from templates
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement

**What happened**:

The templates configure vms to use password authentication, and include a hardcoded passwords. There are two separate security concerns here:

- Setting a default password means that people will start vms with that default configuration. If they are connected to a routeable network, they will rapidly be compromised.
- Passwords of any sort are vulnerable to brute force attempts

**What you expected to happen**:

I would expect the templates to have a template parameter for an SSH public key.

### PR DESCRIPTION
**Pull request by vatsalparekh**
Signed-off-by: Vatsal Parekh <vparekh@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #187 

**Special notes for your reviewer**:

**Release note**:

```release-note
Add a parameter for randomized password
```

### CODE REVIEW

File: templates/_linux.yaml [hunk 1]
```diff
@@ -102,7 +102,7 @@ objects:
         - cloudInitNoCloud:
             userData: |-
               #cloud-config
-              password: {{ password }}
+              password: ${PASSWORD}
```
Comment (ksimon1) [inline templates/_linux.yaml:105]: As we discussed, please change this variable name to ${CLOUD_USER_PASSWORD}

File: templates/_linux.yaml [hunk 2]
```diff
@@ -120,3 +120,7 @@ parameters:
 - name: SRC_PVC_NAMESPACE
   description: Namespace of the source PVC
   value: kubevirt-os-images
+- description: Randomized password
```
Comment (ksimon1) [inline templates/_linux.yaml:123]: Change this to something like: `Randomized password for cloud user`

Comment (vatsalparekh) [reply templates/_linux.yaml]: added

File: templates/_linux.yaml (hunk 1 above)
Comment (vatsalparekh) [reply templates/_linux.yaml]: changed

**Pull request merged**